arkouda/client.py

```diff
@@ -570,30 +577,23 @@ def _json_args_to_str(json_obj: Dict) -> Tuple[int, str]:
             raise TypeError(f"Argument keys are required to be str. Found {type(key)}")
         if isinstance(val, dict):
             raise TypeError("Nested JSON is not currently supported for server messages.")
-        param = {"key": key}
 
         if isinstance(val, pdarray):
-            param["objType"] = "pdarray"
-            param["dtype"] = str(val.dtype)
-            param["val"] = val.name
+            param = ParameterObject(key, ObjectType.PDARRAY, str(val.dtype), val.name)
         elif isinstance(val, Strings):
-            param["objType"] = "SegString"
-            param["dtype"] = "str"
             # empty string if name of String obj is none
-            param["val"] = val.name if val.name else ""
+            name = val.name if val.name else ""
+            param = ParameterObject(key, ObjectType.STRINGS, "str", name)
         elif isinstance(val, list):
-            param["objType"] = "list"
             dtypes = set([p.dtype for p in val])
             if len(dtypes) > 1:
                 t_str = ", ".join(dtypes)
                 raise TypeError(f"List values must be of the same type. Found {t_str}")
-            param["dtype"] = dtypes.pop()
-            param["val"] = json.dumps(val)
+            param = ParameterObject(key, ObjectType.LIST, dtypes.pop(), json.dumps(val))
         else:
-            param["objType"] = type(val).__name__
-            param["dtype"] = type(val).__name__
-            param["val"] = val if isinstance(val, str) else str(val)
-        j.append(json.dumps(param))
+            v = val if isinstance(val, str) else str(val)
+            param = ParameterObject(key, ObjectType.VALUE, type(val).__name__, v)
```

Comment:
I might be wrong but I think now that the we use `ObjectType.VALUE` here instead of `type(val).__name__`, we need to update `getValueAsType` to check that `objType != "value"` instead of `objType != dtype`

https://github.com/Bears-R-Us/arkouda/blob/afd073556545ab32ffabdf7e72511c0e09405bf0/src/Message.chpl#L158-L160

Reply:
You are 100% correct. I update this (and the similar check in `getList`).